src/main/java/com/google/devtools/build/lib/bazel/bzlmod/ModuleExtensionContext.java

```diff
@@ -117,4 +120,54 @@ public StarlarkList<StarlarkBazelModule> getModules() {
   public boolean isDevDependency(TypeCheckedTag tag) {
     return tag.isDevDependency();
   }
+
+  @StarlarkMethod(
+      name = "extension_metadata",
+      doc = "foo",
+      parameters = {
+          @Param(
+              name = "root_module_direct_deps",
+              doc = "foo",
+              positional = false,
+              named = true,
+              defaultValue = "None",
+              allowedTypes = {
+                  @ParamType(type = Sequence.class, generic1 = String.class),
+                  @ParamType(type = NoneType.class)}),
```

Comment:
What do you think about having the string "all" as a possible value for this param to replace `module_ctx.use_all_repos` (similarly for dev deps)?

The obvious downside is the weird typing, but the advantage of this is that we remove some mental burden and _if_ we were to have other fields on ExtensionMetadata later on, it's more composable.

Reply:
yeah. now the question is where to put this as_module_extension thing :)

also notable that use_all_repos is implementable in terms of the "all" thing, but not vice versa. Mostly due to the composability point.

Reply:
Result of an offline discussion: We will go for `"all"`, but realize a wrapper for the simple use case of a WORKSPACE macro either in skylib or Bazel itself.